### PR TITLE
Update Ruby, Node.js, and Node.js dependencies

### PR DESCRIPTION
**Pull request by kaelig**
As I was running `dev up` in this project on my (recently cleanly-reinstalled) laptop, I found that the old, unsupported Ruby (2.3) was causing install issues.

This PR fixes it.

This PR also changes the name of `yarn dist` into `yarn build`, for consistency with other projects such as polaris-icons.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -3,6 +3,8 @@
   "css.validate": false,
   "scss.validate": false,
   "editor.formatOnSave": true,
-  "prettier.stylelintIntegration": true,
-  "prettier.eslintIntegration": true
+  "editor.codeActionsOnSave": {
+    "source.fixAll.eslint": true,
+    "source.fixAll.stylelint": true
```
Comment (BPScott) [inline .vscode/settings.json]: Remove autofixing of stylelint for now, it's got some problems thanks to [upstream bugs with merged but unreleased fixes](https://github.com/stylelint/stylelint/issues/4449). I would say keep it and add js/ts files to the stylelint ignore but that just causes [other problems](https://github.com/stylelint/vscode-stylelint/issues/40)

File: package.json [hunk 2]
```diff
@@ -43,6 +43,7 @@
     "extends": [
       "plugin:shopify/node",
       "plugin:shopify/esnext",
```
Comment (BPScott) [inline package.json]: esnext should be above node as esnext is a "core" config while "node" is an extension, per https://github.com/shopify/eslint-plugin-shopify#provided-configurations

File: __tests__/package.json.test.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 describe('package.json', () => {
```
Comment (BPScott) [inline __tests__/package.json.test.js]: IIRC tests are found if they're in the `__tests__` directory or if they end in `.test.js`, We tend to just do the latter.

Can you rename this to `tests/package.json.test.js`

File: package.json (hunk 2 above)
Comment (kaelig) [reply package.json]: Thanks! I had no idea plugins were order-sensitive.

**Pull request merged**